src/storage/src/hummock/sstable/delete_range_aggregator.rs

```diff
@@ -187,6 +195,22 @@ impl<I: DeleteRangeIterator> DeleteRangeAggregator<I> {
         }
     }
 
+    fn add_all_overlap_range(&mut self, include_key: &[u8]) {
+        while self.inner.is_valid() && self.inner.start_user_key().le(include_key) {
+            let sequence = self.inner.current_epoch();
+            if sequence > self.watermark || self.inner.end_user_key().le(include_key) {
+                self.inner.next();
+                continue;
+            }
+            self.end_user_key_index.push(SortedBoundary {
+                user_key: self.inner.end_user_key().to_vec(),
+                sequence,
+            });
+            self.epoch_index.insert(sequence);
+            self.inner.next();
+        }
+    }
+
     /// Check whether the target-key is deleted by some range-tombstone. Target-key must be given
     /// in order.
     pub fn should_delete(&mut self, target_key: &[u8], epoch: HummockEpoch) -> bool {
```

Comment:
I think we can check `self.inner.is_valid()` in the very beginning and early return false directly. This is the normal case when we have no delete range tombstones in SST.

Reply:
Btw, I think the following check should be:
```
        if epoch > self.watermark {
            return false;
        }
```

Using `>=` may also be correct since we assume that keys covered in a range tombstone can be written using the same epoch as the tombstone epoch. But I think using `>` is clearer.

Reply:
No.... We can not check `self.inner.is_valid` because we only care about the epoch of which delete-range include this key.